quinn-h3/src/client.rs

```diff
@@ -48,21 +45,14 @@ impl Builder {
         self
     }
 
-    pub fn bind<T: ToSocketAddrs>(
-        self,
-        addr: T,
-    ) -> Result<(EndpointDriver, Client), EndpointError> {
-        let (endpoint_driver, endpoint, _) = self.endpoint.bind(addr)?;
-        Ok((
-            endpoint_driver,
-            Client {
-                endpoint,
-                settings: self.settings,
-                log: self
-                    .log
-                    .unwrap_or_else(|| Logger::root(slog::Discard, o!())),
-            },
-        ))
+    pub fn endpoint(self, endpoint: Endpoint) -> Client {
```

Comment:
Nice change! I like that we can now share endpoints between multiple types of outgoing connections.